### PR TITLE
Hip clang CI jobs added

### PR DESCRIPTION
**Pull request by akilaMD**
- Adds CentOS and hip-clang jobs
- Needs to pass CI

### CODE REVIEW

**Comment by akilaMD** [comment]
Only failures are General SCM-related. This should be merged into develop

**Comment by akilaMD** [comment]
Passed CI, and tested it on jenkins-rocm-9 which has 2 GPUs. Should be good to merge.

File: docker/dockerfile-install-centos [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Parameters related to building hipcub
+ARG base_image
+
+FROM ${base_image}
+LABEL maintainer="saad.rahim@amd.com"
```
Comment (saadrahim) [inline docker/dockerfile-install-centos]: We should no longer use any personal email addresses. Please change to hipcub-maintainer@amd.com

File: docker/dockerfile-build-centos [hunk 2]
```diff
@@ -0,0 +1,45 @@
+# Parameters related to building hipCUB
+ARG base_image
+
+FROM ${base_image}
+LABEL maintainer="saad.rahim@amd.com"
```
Comment (saadrahim) [inline docker/dockerfile-build-centos]: same here as below

Comment (akilaMD) [reply docker/dockerfile-build-centos]: Latest commit updates the maintainer.

**Pull request merged**